### DISCUSSION TITLE
V1 idea: globally unique channel ID

### DISCUSSION THREAD

**Discussion by wkalt** [foxglove/mcap#1103] [General]
If a recording process is rotating mcap files, and a downstream process wants to merge them together, today there is no sane way to ensure that channels are aligned properly across log rotations. Just matching on topic/schema may not be sufficient, nor is a hash of channel metadata adequate. This could be solved by assigning a UUID value to the channel record and using the same UUID across rotations. This would probably require a new field to accomplish, hence tagging "v1".

**Comment by james-rms** [comment]
> If a recording process is rotating mcap files, today there is no sane way to ensure that channels are aligned properly across log rotations

I don't this problem statement is well formed at the moment. As stated, the recording process could persist the channel IDs used across rotations, and the merging process could use the channel ID as the merge key.

**Reply by defunctzombie** [reply]
> merging process could use the channel ID as the merge key

The merging process cannot assume that a channel ID in one file represents the same channel ID in another file. The IDs are local to a file and the spec does not call for anything about the IDs across files.
